src/Persistence.Memory/InMemoryAggregateRepository.cs

```diff
@@ -1,36 +1,55 @@
-﻿using System;
-using System.Collections.Concurrent;
-using System.Collections.Generic;
-using System.Linq;
-using CR.AggregateRepository.Core;
-using CR.AggregateRepository.Core.Exceptions;
+﻿// <copyright file="InMemoryAggregateRepository.cs" company="Cognisant">
+// Copyright (c) Cognisant. All rights reserved.
+// </copyright>
 
 namespace CR.AggregateRepository.Persistence.Memory
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Core;
+    using Core.Exceptions;
+
+    /// <inheritdoc />
+    /// <summary>
+    /// A simple in-memory aggregate repository implementation which reads events from, and writes them to, a <see cref="ConcurrentDictionary{TKey,TValue}"/>.
+    /// </summary>
     public class InMemoryAggregateRepository : IAggregateRepository
     {
-        public readonly ConcurrentDictionary<object,List<object>> EventStore = new ConcurrentDictionary<object, List<object>>();
-        
+        /// <summary>
+        /// In-memory event storage.
+        /// </summary>
+#pragma warning disable SA1401 // Fields should be private
+        public readonly ConcurrentDictionary<object, List<object>> EventStore = new ConcurrentDictionary<object, List<object>>();
+#pragma warning restore SA1401 // Fields should be private
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InMemoryAggregateRepository"/> class with an empty event store.
+        /// </summary>
         public InMemoryAggregateRepository()
         {
-            
         }
 
-        public InMemoryAggregateRepository(Dictionary<object, List<object>> initialEvents)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InMemoryAggregateRepository"/> class, with the provided streams (and events) contained in the provided <see cref="IDictionary{TKey,TValue}"/>.
+        /// </summary>
+        /// <param name="initialEvents">A dictionary of initial events added when itializing the instance.</param>
+        // ReSharper disable once UnusedMember.Global
+        public InMemoryAggregateRepository(IDictionary<object, List<object>> initialEvents)
         {
             foreach (var item in initialEvents)
             {
-                EventStore.TryAdd(item.Key, item.Value);
+                EventStore.TryAdd(item.Key, new List<object>(item.Value));
             }
         }
 
+        /// <inheritdoc />
```

Comment:
We say `See implementation specific documentation for the circumstances in which this may be thrown.` on the base class for this but haven't actually specified on the implementation when it would be thrown.